interfaces/builtin/steam_support.go

```diff
@@ -0,0 +1,201 @@
+# Steam Client / API   #
+########################
+
+# Allow Valve IPC to work for the multiprocess client architecture.
+# "Server" component will write/create/lock, clients (i.e. running games in
+# the Steam API) will mmap the object.
+owner /{dev,run}/shm/u*-ValveIPCSharedObj* mrwk,
+
+# Linux Steam Integration config mask system
+owner /{dev,run}/shm/u*-LinuxSteamIntegration.unity3d.* rw,
+
+# i.e. /dev/shm/#539043
+owner /{dev,run}/shm/#* mrw,
+
+# Further required pipe devices (i.e. /dev/shm/u1000-Shm_125376fc)
+owner /{dev,run}/shm/u*-Shm_* mrw,
+
+# Steam Web Store IPC (i.e. /dev/shm/ValveIPCSHM_1000)
+owner /{dev,run}/shm/ValveIPCSHM_* rw,
+
+# For old Source games like L4D2
+owner /{dev,run}/shm/{,.}org.chromium.Chromium.shmem.libcef_* mrwk,
+
+
+########################
+#    Audio Support     #
+########################
+
+# Base runtime can provide HRTF/OpenAL implementation for games
+/usr/share/openal/hrtf/ r,
+/usr/share/openal/hrtf/*.mhr mr,
+/usr/share/openal/presets/ r,
+/usr/share/openal/hrtf/*.{ambdec,txt} mr,
+
+# solus-runtime-gaming uses a stateless alsa configuration which will
+# always use PulseAudio if possible.
+/usr/share/alsa/alsa.conf r,
+/usr/share/defaults/alsa/asound.conf r,
+/usr/share/alsa/alsa.conf.d/ r,
+/usr/share/alsa/alsa.conf.d/** r,
+/usr/share/alsa/cards/** r,
+/usr/share/alsa/pcm/** r,
+
+# PCM Access
+/sys/devices/pci**/sound/**/pcm**/pcm_class r,
+
+########################
+#  Mono Game Support   #
+########################
+
+# i.e. /dev/shm/mono-shared-1000-shared_data-ironhide-Linux-x86_64-328-12-0
+owner /{dev,run}/shm/mono-shared-*-shared_{data,fileshare}-* mrw,
+
+# i.e. /dev/shm/mono.16965
+owner /{dev,run}/shm/mono.@{pid} mrw,
+
+
+########################
+#     GPU Access       #
+########################
+
+# Games can and will attempt to map /dev/nvidia* (specifically Unity)
+/dev/nvidiactl mr,
+/dev/nvidia[0-9] mr,
+
+########################
+#    Vulkan Support    #
+########################
+
+# Base runtime can contain the vulkan loader itself
+/usr/share/vulkan/explicit_layer.d/ r,
+/usr/share/vulkan/explicit_layer.d/** r,
+
+# Base runtime can contain mesalib definitions for Vulkan ICD files
+/usr/share/vulkan/icd.d/ r,
+/usr/share/vulkan/icd.d/** r,
+
+########################
+#      Disks/Media     #
+########################
+
+# Steam requires +x permissions on the partitions and will perform
+# such a test:
+# sh: /run/media/bigdisk/games//steamapps/.steam_exec_test.sh: /bin/sh: bad interpreter: Permission denied
+/{,run/}media/**/.steam_exec_test.sh ixmrw,
+
+# Libraries and executables on other partitions require map + execute permissions
+/{,run/}media/**/steamapps/common/** ixm,
+
+########################
+# Udev / PCI Support   #
+########################
+
+# Steam Controller
+/run/udev/data/c247:[0-9]* r,
+/run/udev/data/c248:[0-9]* r,
+
+# Allow thread-synchronisation to work in the Steam client
+/sys/devices/pci[0-9]*/**/{class,enable,irq,resource*} r,
```

Comment:
The `**` seems broad. Is this for the controller? If so then moving to joystick interface and using:
```
# Allow thread-synchronisation
/sys/devices/pci[0-9a-f]*/**/input[0-9]*/**/{class,enable,irq,resource*} r,
```

It would be nice to see the denial that triggered this rule.